### ISSUE TITLE
Errors when activating API Gateway’s cache on AWS: no CORS headers returned

### ISSUE THREAD

**Issue by mkjsix** [serverless/serverless#4270]
# This is a Bug Report

## Description

We have a set of different APIs created by serverless (>= 1.18). They work nicely (with CORS) in production since months. Now we need to [activate the API Gateway's cache](http://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-caching.html) but after that the API gives errors. Specifically I see this in browsers' Javascript console:

`No 'Access-Control-Allow-Origin' header is present on the requested resource.`

When we press the "Flush cache" button in the API Gateway console, the first call to the API works (it doesn't go through the CloudFront cache), but as soon as the resource gets cached again, we immediately get the mentioned error again.

In summary, it seems that when activating the API Gateway cache, responses coming from CloudFront's cache "lose" the CORS headers.

For the moment we have disabled all API Gateway caches and everything is working back to normal, but this is going to create performance issues on the backend as usage of the API is increasing.

Has anybody else experienced this behavior or are we missing any configuration on serverless side? Is there any workaround?

## Additional Data

* ***Serverless Framework Version you're using***: 1.21.1
* ***Operating System***:
* ***Stack Trace***:
* ***Provider Error messages***:

**Comment by fmontada** [comment]
Hi mkjsix we are having the same issue, did you find any solution ?

**Comment by mkjsix** [comment]
@fmontada we haven't find any solution, let's say we are still on hold with this caching problem. Meanwhile we have created other APIs but without using the Serverless framework, to avoid this issue.

**Comment by montadaf** [comment]
@mkjsix when we flush chance using either of the clients below, it work properly 

https://docs.aws.amazon.com/cli/latest/reference/apigateway/flush-stage-cache.html

https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/apigateway/AmazonApiGatewayClient.html

**Comment by montadaf** [comment]
@mkjsix in our flow we flush the cache every time a user add or update a Product, then using any of the client I mention we invalida the cache
